nautobot/docs/development/getting-started.md

```diff
@@ -22,52 +36,158 @@ LICENSE.txt      development  mkdocs.yml  pyproject.toml           upgrade.sh
+Now that you've got the latest upstream changes, create your branch. It's convention to always prefix your branch name with your GitHub username, separated by hyphens. For example:
+
+```no-highlight
+$ git checkout -b yourusername-myfeature
+```
 
 ## Enabling Pre-Commit Hooks
 
-Nautobot ships with a [git pre-commit hook](https://githooks.com/) script that automatically checks for style compliance and missing database migrations prior to committing changes. This helps avoid erroneous commits that result in CI test failures. You are encouraged to enable it by creating a link to `scripts/git-hooks/pre-commit`:
+Nautobot ships with a [Git pre-commit hook](https://githooks.com/) script that automatically checks for style compliance and missing database migrations prior to committing changes. This helps avoid erroneous commits that result in CI test failures. You are encouraged to enable it by creating a link to `scripts/git-hooks/pre-commit`:
 
 ```no-highlight
 $ cd .git/hooks/
 $ ln -s ../../scripts/git-hooks/pre-commit
 ```
 
-## Setting up a Development Environment
+## Setting up your Development Environment
+
+Getting started with Nautobot development is pretty straightforward, and should feel very familiar to anyone with Django development experience. We can recommend either a [Docker Compose workflow](#docker-compose-workflow) (if you don't want to install dependencies such as PostgreSQL and Redis directly onto your system) or a [Python virtual environment workflow](#python-virtual-environment-workflow).
+
+### Docker Compose Workflow
+
+This workflow uses [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/) and assumes that you have them installed.
 
-Getting started with Nautobot development is pretty straightforward, and should feel very familiar to anyone with Django development experience. We can recommend either a [Docker Compose workflow](#docker-development-environment-workflow) (if you don't want to install dependencies such as PostgreSQL and Redis directly onto your system) or a [Python virtual environment workflow](#python-virtual-environment-workflow).
+For the Docker Compose workflow, Nautobot uses [Invoke](http://docs.pyinvoke.org/en/latest/index.html) as a replacement for Make. Invoke was chosen because it is less arcane than make. Instead of a `Makefile`, Invoke reads the `tasks.py` in the project root.
 
-### Docker Development Environment Workflow
+#### Install Invoke
 
-A development environment can be easily started up from the root of the project by installing the [Invoke](http://docs.pyinvoke.org/en/latest/invoke.html) Python library and then using the following commands:
+Because it is used to execute all common Docker workflow tasks, Invoke must be installed globally for your user environment.
```

Comment:
We shouldn't be recommending to globally install any Python packages, not even Invoke - we should either:
- direct the user to do a user install (`pip install --user invoke`) and ensure that their `$PATH` is set correctly
- or direct the user to create a virtualenv and install Invoke there,
- or install Poetry and use Poetry to create a virtualenv containing Invoke.

Reply:
`pip install --user` is the default these days unless you explicitly install as root or a user that has permissions to modify the system-installed version.

Really the goal is "Invoke must be installed for use outside of virtualenvs" similar to that of Poetry. And then we should describe them in the same way.

Reply:
OK. I was thrown by the "installed globally" wording.

Reply:
Yeah, I'll flesh it out.